api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -36,7 +36,7 @@ public void setDao(PatientQueueingDao dao) {
     public PatientQueue savePatientQue(PatientQueue patientQueue) {
         PatientQueue currentQueue = dao.getIncompletePatientQueue(patientQueue.getPatient(), patientQueue.getLocationTo());
 
-        if (currentQueue != null) {
+        if (currentQueue != null && !patientQueue.equals(currentQueue)) {
```

Comment:
Do you have a test to catch this bug?

Reply:
@dkayiwa  yes. I added an assert statement to one of the tests to ensure that this is tested.